src/Utils.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import MetaProperty from "./MetaProperty";
+import { PathUtils } from "./PathUtils";
+import Constants from "./Constants";
+
+export class Utils {
+
+    /**
+     * Returns true if the editor is in Edit Mode
+     *
+     * @returns {Boolean} the result of the check of the Editor mode
+     */
+    public static isEditMode(): boolean {
+        const aemMode = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE);
+        if (aemMode != null) {
+            return aemMode === Constants.AEM_MODE_EDIT;
+        }
+        else {
+            if (PathUtils.isBrowser()) {
+                const param = Utils.getEditParam();
+                return param === Constants.AEM_MODE_EDIT;
+            }
+            else {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Returns query parameter value for aem mode.
+     *
+     * @returns {String | null} the result of the check of the Editor mode
+     */
+    public static getEditParam() : string | null {
+        return new URL(document.location.href).searchParams.get(Constants.AEM_MODE_KEY);
```

Comment:
`new URL` can potentially throw exception, so should not be called without try/catch